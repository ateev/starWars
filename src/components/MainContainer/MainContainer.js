import React, { Component } from 'react';
import { readBrowserCookie } from '../../helpers/cookieHandler.js';
import { checkCreds } from '../../actions/userActions.js';
import Login from '../login/login.js';
import PlanetsContainer from '../planetsContainer/planetsContainer';
import { getUserId } from '../../loginManager.js';
import { connect } from 'react-redux';

class MainContainer extends Component {
  loginUser = (username, password) => {
    this.context.store.dispatch(checkCreds(username, password));
  }
  render() {
    const userName = readBrowserCookie('LoggedIn username')[0];
    let appBlock;
    if(typeof userName !== 'undefined') {
      appBlock = <PlanetsContainer />;
    } else {
      appBlock = <Login submitCallback={ this.loginUser } />
    }
    return (
      <div className="main-container">
        { appBlock }
      </div>
    );
  }
}

MainContainer.contextTypes = {
  store: React.PropTypes.object,
};

function mapStateToProps(state) {
  return {
    user: state.user,
  };
}

export default connect(mapStateToProps)(MainContainer);
