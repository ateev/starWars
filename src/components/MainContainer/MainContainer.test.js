import React from 'react';
import configureMockStore from 'redux-mock-store';
import { shallow, mount } from 'enzyme';
import MainContainer from './MainContainer.js';
import { Provider } from 'react-redux';

const mockStore = configureMockStore();
const initialState = {};
const store = mockStore(initialState);

describe('<MainContainer />', () => {
    it('renders without crashing', () => {
        const component = shallow(
            <Provider store={store}>
                <MainContainer />
            </Provider>
        );
        expect(component).toMatchSnapshot();
    });
})
