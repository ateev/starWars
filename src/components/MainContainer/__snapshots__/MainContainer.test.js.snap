exports[`<MainContainer /> renders without crashing 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Connect(MainContainer) />,
  "nodes": Array [
    <Connect(MainContainer) />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Provider
        store={
                Object {
                        "clearActions": [Function],
                        "dispatch": [Function],
                        "getActions": [Function],
                        "getState": [Function],
                        "replaceReducer": [Function],
                        "subscribe": [Function],
                      }
        }>
        <Connect(MainContainer) />
</Provider>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Provider {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": <Connect(MainContainer) />,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
        },
        "refs": Object {},
        "state": null,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Connect(MainContainer) />,
        "_debugID": 2,
        "_renderedOutput": <Connect(MainContainer) />,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Provider
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }>
    <Connect(MainContainer) />
</Provider>,
}
`;
